lib/semantics/check-omp-structure.h

```diff
@@ -108,41 +106,43 @@ class OmpStructureChecker : public virtual BaseChecker {
   void Enter(const parser::OmpReductionClause &);
   void Enter(const parser::OmpScheduleClause &);
 
-  bool HasInvalidCloselyNestedRegion(
-      const parser::CharBlock &, const OmpDirectiveEnumType &);
-  void CheckAllowed(const OmpClauseEnum &);
-
 private:
   struct OmpContext {
     parser::CharBlock currentDirectiveSource{nullptr};
     parser::CharBlock currentClauseSource{nullptr};
-    OmpDirectiveEnum currentDirectiveEnum;
-    OmpClauseEnumType allowedClauses;
-    OmpClauseEnumType allowedOnceClauses;
-    OmpClauseEnumType seenClauses;
+    OmpDirective currentDirectiveEnum;
```

Comment:
I suggest the names `directiveSource`, `clauseSource`, and `directive` for these first three data members. It should be clear they are "current" if they come from the top of the stack. And the others one the stack aren't current.

Reply:
okay!

Reply:
Addressed in [f10920a](f10920a83170b9350b7efd8a164f9b0f70ac46dc).